### PR TITLE
Added mutability test for user functions that save documents

### PR DESCRIPTION
**Pull request by shivanshrustagi**
CBG-0000

Describe your PR here...
- Added the test for all the mutability cases
   - If a function with a mutating value of true calls another function w/ a mutating value of false, the value of mutating is false.
   - If a function with a mutating value of false calls another function w/ a mutating value of true, the value of mutating is false.
   - If a function with a mutating value of false calls another function w/ a mutating value of false, the value of mutating is false.
   - If a function with a mutating value of true calls another function w/ a mutating value of true, the value of mutating is true.

## Pre-review checklist
- [ ] Removed debug logging (`fmt.Print`, `log.Print`, ...)
- [ ] Logging sensitive data? Make sure it's tagged (e.g. `base.UD(docID)`, `base.MD(dbName)`)
- [ ] Updated relevant information in the API specifications (such as endpoint descriptions, schemas, ...) in `docs/api`

## Dependencies (if applicable)
- [ ] Link upstream PRs
- [ ] Update Go module dependencies when merged

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [ ] `GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/000/

### CODE REVIEW

File: rest/functionsapitest/user_functions_test.go [hunk 1]
```diff
@@ -1079,3 +1079,113 @@ func testUserQueriesAsUser(t *testing.T, rt *rest.RestTester) {
...
+			Args:     []string{"doc", "docID", "funcName"},
+			Allow:    &functions.Allow{Channels: []string{"*"}},
+			Mutating: false,
+		},
+	},
+}
+
+func TestFunctionMutability(t *testing.T) {
+	rt := rest.NewRestTesterForUserQueries(t, rest.DbConfig{})
+	defer rt.Close()
```
Comment (jineshparakh) [inline rest/functionsapitest/user_functions_test.go:1125]: You'll have to add a check for rt being nil
```
if rt == nil {
    return
}
```

Comment (jineshparakh) [inline rest/functionsapitest/user_functions_test.go:1128]: You'll have to add a check for rt being nil.
```
if rt == nil { 
    return 
}
```

File: rest/functionsapitest/user_functions_test.go [hunk 2]
```diff
@@ -1079,3 +1079,116 @@ func testUserQueriesAsUser(t *testing.T, rt *rest.RestTester) {
...
+		"docID": "Test123",
+		"funcName": "%s"
+	}`
+	var putFuncName string
+	var callerFuncName string
+
+	response := rt.SendAdminRequest("PUT", "/db/_config/functions", string(request))
+	assert.Equal(t, 200, response.Result().StatusCode)
+
+	//function with a mutating value of true calls another function w/ a mutating value of false
```
Comment (moshaad7) [inline rest/functionsapitest/user_functions_test.go]: Redundancy: comment and test title conveying the same thing 
U can remove the comment

File: rest/functionsapitest/user_functions_test.go [hunk 3]
```diff
@@ -1079,3 +1079,116 @@ func testUserQueriesAsUser(t *testing.T, rt *rest.RestTester) {
...
+	t.Run("Func with mutating True calls another function with a mutating value of False", func(t *testing.T) {
+		putFuncName = "putDocMutabilityFalse"
+		callerFuncName = "callerMutabilityTrue"
+		response := rt.SendAdminRequest("POST", fmt.Sprintf("/db/_function/%s", callerFuncName), fmt.Sprintf(body, putFuncName))
+		assert.Equal(t, http.StatusForbidden, response.Result().StatusCode)
+	})
+
+	//function with a mutating value of false calls another function w/ a mutating value of true
```
Comment (moshaad7) [inline rest/functionsapitest/user_functions_test.go]: redundancy (comment and test title)

File: rest/functionsapitest/user_functions_test.go [hunk 4]
```diff
@@ -1079,3 +1079,116 @@ func testUserQueriesAsUser(t *testing.T, rt *rest.RestTester) {
...
+	t.Run("Func with mutating False calls another function with a mutating value of True", func(t *testing.T) {
+		putFuncName = "putDocMutabilityTrue"
+		callerFuncName = "callerMutabilityFalse"
+		response := rt.SendAdminRequest("POST", fmt.Sprintf("/db/_function/%s", callerFuncName), fmt.Sprintf(body, putFuncName))
+		assert.Equal(t, http.StatusForbidden, response.Result().StatusCode)
+	})
+
+	//function with a mutating value of false calls another function w/ a mutating value of false
```
Comment (moshaad7) [inline rest/functionsapitest/user_functions_test.go]: redundancy (comment and test title)

**Review by moshaad7 — COMMENTED** [review]
minor comments

**Pull request merged**